Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp

```diff
@@ -1932,6 +1929,11 @@ void WebsiteDataStore::propagateSettingUpdatesToNetworkProcess()
     if (m_isOptInCookiePartitioningEnabled != enabled) {
         m_isOptInCookiePartitioningEnabled = enabled;
         networkProcess->send(Messages::NetworkProcess::SetOptInCookiePartitioningEnabled(sessionID(), enabled), 0);
+
+        if (m_isOptInCookiePartitioningEnabled && m_thirdPartyCookieBlockingMode == WebCore::ThirdPartyCookieBlockingMode::All)
+            setThirdPartyCookieBlockingMode(WebCore::ThirdPartyCookieBlockingMode::AllExceptPartitioned, []() { });
+        else if (!m_isOptInCookiePartitioningEnabled && m_thirdPartyCookieBlockingMode == WebCore::ThirdPartyCookieBlockingMode::AllExceptPartitioned)
+            setThirdPartyCookieBlockingMode(WebCore::ThirdPartyCookieBlockingMode::All, []() { });
```

Comment:
Can you add log for changing `ThirdPartyCookieBlockingMode` based on `m_isOptInCookiePartitioningEnabled` value? It seems the relationship is kind of implicit to client.

Reply:
Yeah, good idea.